datamonster_api/lib/data_group.py

```diff
@@ -35,6 +46,46 @@ def get_details(self):
         """
         return self.dm.get_data_group_details(self.id)
 
+    def start_data_refresh(self, data_frame):
+        if sum(data_frame.memory_usage()) > data_frame_cutoff_size:
+            raise DataMonsterError('Data Too Large. Data Groups can be refreshed with data < 64 MB.')
+
+        avro_file = BytesIO(dataframe_to_avro_bytes(data_frame, 'upload_data', 'com.adaptivemgmt.upload'))
+        if avro_file.getbuffer().nbytes > max_file_size:
+            raise DataMonsterError('Data Too Large. Data Groups can be refreshed with data < 64 MB.')
+
+        files = {'avro_file': avro_file}
+        headers = {'Accept': 'avro/binary'}
+        try:
+            return self.dm.client.post(self._get_refresh_url(), {}, headers=headers, files=files)
+        except Exception:
+            raise DataMonsterError('Unknown problem refreshing data. Please contact DataMonster Customer Service.')
```

Comment:
we don't want just the DM error to propagate through? if it's not a 500 will those be logged so we can debug?

Reply:
seems like we're just masking any issues

Reply:
I figured the API side should be the thing that decides what the user-facing message is, but I'm open to other suggestions.

Reply:
it will be hard to debug user issues when they say oh we hit this error, but then we have to do extra work to try and reproduce. instead if we see what the server error is, perhaps we can debug a lot faster

Reply:
i don't have a particularly strong sway here, i just think it will be harder to debug stuff in the future with a generic catchall message